UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API/Validation/D365PayloadInputParametersValidation.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Linq;
+using UKHO.ExternalNotificationService.Common.Models.Request;
+
+namespace UKHO.ExternalNotificationService.API.Validation
+{
+    public static class D365PayloadInputParametersValidation
+    {
+        public static bool IsValidAttribute(this D365Payload payload,string postEntityImageKey, string attributeKey)
+        {
+            if (IgnoreRuleForValidation(payload))
+            { return true; }
+            try
+            {
+                var inputParameter = payload.InputParameters.Single();
+                var postEntityImage = payload.PostEntityImages.SingleOrDefault(i => i.Key == postEntityImageKey);
+                var attributes = inputParameter.Value.Attributes.Concat(postEntityImage?.Value?.Attributes ?? Array.Empty<D365Attribute>());
+
+                var validAttributeKey = attributes.FirstOrDefault(a => a.Key == attributeKey);
+                return (validAttributeKey != null && !string.IsNullOrWhiteSpace(attributes.FirstOrDefault(a => a.Key == attributeKey).Value.ToString()));
+            }
+            catch (Exception)
+            {
+                return false;
+            }
+        }
+        public static bool IsValidFormatted(this D365Payload payload, string postEntityImageKey, string formattedKey)
+        {
+            if (IgnoreRuleForValidation(payload))
+            { return true; }
+            try
+            {
+                var inputParameter = payload.InputParameters.Single();
```

Comment:
I feel like this is very similar code to the subscription service. Is there some method, perhaps to retrieve values from a D365 payload, that can be centralized.

Reply:
Thanks for highlighting, added centralized class.